src/Fusion/Plugin.hs

```diff
@@ -272,19 +352,45 @@ markInline reportMode failIt transform guts = do
         --mapM_ (\(b, expr) -> transformBind dflags anns (NonRec b expr)) bs
         return bndr
 
+reportAnns :: ReportMode -> ModGuts -> CoreM ModGuts
+reportAnns reportMode guts = do
+    putMsgS $ "fusion-plugin: Checking presence of annotated types..."
+    dflags <- getDynFlags
+    anns <- getAnnotations deserializeWithData guts
+    if (anyUFM (any (== Fuse)) anns)
+    then bindsOnlyPass (mapM (transformBind dflags anns)) guts
+    else return guts
+  where
+    transformBind :: DynFlags -> UniqFM [Fuse] -> CoreBind -> CoreM CoreBind
+    transformBind dflags anns bind@(NonRec b _) = do
+        let parentName = showSDoc dflags (ppr b)
+        when ("main" `DL.isPrefixOf` parentName) $ do
```

Comment:
What are we doing here? Why are we looking for "main" being the prefix of a bndr?

Reply:
We are looking for code that is reachable from "main" so that we do not report any other top level bindings which are not being used or which may not even be fused. Is there a better way to do this?

Reply:
I am assuming you mean the main function here, but complete stream fusion (construction & elimination both) can happen in other functions that are reachable from main and yet not inlined, i.e. we shall have modules without a main function in which stream fusion is supposed to happen. 

I do not think this filtering is useful.

Reply:
yes, and I was thinking having another annotation `Fused` for that instead of using this filtering. The programmer can annotate functions that are supposed be fused and we will report in those functions. Here we are assuming that only `main` is annotated as `Fused` which is only one of fused functions.

I think we can remove this and let the noise occur for now and introduce a `Fused` annotation later. Or do you have some other idea?

Reply:
Having the `Fused` annotation on functions looks like a good idea. Yes for now we should remove this and let the noise occur I think.

I can remove it and merge this PR if you're busy on other things.

Reply:
Sure. Go ahead. Thanks!